timemachine/md/hrex.py

```diff
@@ -0,0 +1,247 @@
+from dataclasses import dataclass
+from typing import Callable, Generic, List, NewType, Optional, Sequence, Tuple, TypeVar
+
+import numpy as np
+from numpy.typing import NDArray
+
+from timemachine.md.moves import MetropolisHastingsMove, MixtureOfMoves
+from timemachine.utils import batches
+
+_Replica = TypeVar("_Replica")
+
+StateIdx = NewType("StateIdx", int)
+ReplicaIdx = NewType("ReplicaIdx", int)
+
+
+@dataclass
```

Comment:
nit: why is this a dataclass?  (it doesn't seem to declare any internals in the usual dataclass style)

Reply:
No strong preference; I've tended to prefer dataclasses for classes with a trivial `__init__` method. Switching to a regular class here would allow preventing mutation of `replicas` and `replica_idx_by_state`, which might be preferable

Reply:
Ah, sorry, I misread the location of this comment and thought it applied to the `Hrex` class. I didn't intend `NeighborSwapMove` to be a dataclass; this is left over from a previous design. Will remove.

Reply:
Removed in https://github.com/proteneer/timemachine/pull/1128/commits/46707c9c63b3c38ae2f594dba1f0e40dbaf3911d